activerecord/test/cases/migration/references_foreign_key_test.rb

```diff
@@ -105,6 +105,44 @@ class ReferencesForeignKeyTest < ActiveRecord::TestCase
           @connection.remove_reference :testings, :testing_parent, foreign_key: true
         end
       end
+
+      test "foreign key column can be added for singular table" do
+        begin
+          original_pluralize_table_names = ActiveRecord::Base.pluralize_table_names
+          ActiveRecord::Base.pluralize_table_names = false
+          @connection.create_table :testing
+          @connection.change_table :testing_parents do |t|
+            t.references :testing, foreign_key: true
+          end
+
+          fk = @connection.foreign_keys("testing_parents").first
+          assert_equal "testing_parents", fk.from_table
+          assert_equal "testing", fk.to_table
+        ensure
+          ActiveRecord::Base.pluralize_table_names = original_pluralize_table_names
+          @connection.drop_table "testing_parents"
```

Comment:
I have to drop testing parent table here because I'm adding foreign key to it. If you don't drop it you can't drop testing table too.